docs/dev_install.rst

```diff
@@ -11,48 +11,83 @@ For precise build instructions on Windows, please check out `appveyor.yml <https
 
 Note that on Linux/OS X, ``clang`` is the only supported compiler for compiling the Taichi compiler. On Windows only MSVC supported.
 
-Installing Depedencies
-----------------------
+Installing Dependencies
+-----------------------
 
 - Make sure you are using Python 3.6/3.7/3.8
-- Execute
+- Install Python dependencies:
 
   .. code-block:: bash
 
-    python3 -m pip install --user setuptools astor pybind11 numpy
-    python3 -m pip install --user pytest pytest-rerunfailures pytest-xdist pytest-cov
-    python3 -m pip install --user GitPython coverage autograd Pillow yapf
-    python3 -m pip install --user scipy psutil opencv-python astpretty
+    python3 -m pip install --user setuptools astpretty astor pybind11
+    python3 -m pip install --user pytest pytest-rerunfailures pytest-xdist yapf Pillow
+    python3 -m pip install --user numpy GitPython coverage colorama autograd
 
-* (If on Ubuntu) Execute ``sudo apt install libtinfo-dev clang-8``. (``clang-7`` should work as well).
+- Make sure you have ``clang`` with version >= 7
 
-* (If on other Linux distributions) Please build clang 8.0.1 from source:
+  * On Windows: Download ``clang-8`` via `this link <https://releases.llvm.org/8.0.0/LLVM-8.0.0-win64.exe>`_.
+    Make sure you add the ``bin`` folder containing ``clang.exe`` to the ``PATH`` environment variable.
 
-  .. code-block:: bash
+  * On OS X: you don't need to do anything.
 
-    wget https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/cfe-8.0.1.src.tar.xz
-    tar xvJf cfe-8.0.1.src.tar.xz
-    cd cfe-8.0.1.src
-    mkdir build
-    cd build
-    cmake ..
-    make -j 8
-    sudo make install
+  * On Ubuntu, execute ``sudo apt install libtinfo-dev clang-8``.
 
+  * On other Linux distributions, please build clang 8.0.1 from source:
 
-- Make sure you have LLVM 8.0.1 built from source. To do so:
+    .. code-block:: bash
 
-  .. code-block:: bash
+        wget https://github.com/llvm/llvm-project/releases/download/llvmorg-8.0.1/cfe-8.0.1.src.tar.xz
+        tar xvJf cfe-8.0.1.src.tar.xz
+        cd cfe-8.0.1.src
+        mkdir build
+        cd build
+        cmake ..
+        make -j 8
+        sudo make install
+
+
+- Make sure you have LLVM 8.0.1/10.0.0. Note that Taichi uses a customized LLVM so the pre-built binaries from the LLVM official website or other sources probably doesn't work.
+  Here we provide LLVM 8.0.1 binaries customized for Taichi, which may or may not work depending on your system environment:
```

Comment:
Although not really related to this PR, what kind of customizations happen in the pre-built binaries? I believe this is related to tasks in #655.

Reply:
Basically the additional CMake options: https://github.com/taichi-dev/taichi/pull/1202/files#diff-97322f732b832df07874b286a638a2a1R66

I guess RTTI is what leads to incompatibilities (although I haven't confirmed).